playground/src/lib/editor-ops/definitions-provider.ts

```diff
@@ -0,0 +1,91 @@
+import * as monaco from 'monaco-editor';
+import { LANG } from '../constants';
+import { getDeclarationNameFromId, getNextSpanStart, getPipelineElementByOffset } from '../nushell';
+import { moveSpanByOffset, spanToRange } from '../utils';
+
+export const addDefinitionProvider = (editor: monaco.editor.IStandaloneCodeEditor): monaco.IDisposable => {
+  const fakeUri = monaco.Uri.parse("inmemory://fake");
+  const fakeModel = monaco.editor.createModel("", LANG, fakeUri);
+  // @ts-ignore this is a monkey patch
+  editor._codeEditorService.openCodeEditor = async (
```

Comment:
how did you find how should you monkey patch, add a URL to the source code so it will be possible to understand what the fuck is going on here